### PR TITLE
Add filters and mock services

### PR DESCRIPTION
**Pull request by trane**
This will merge all current work from our disparate branches.

### CODE REVIEW

File: borderpatrol-core/src/main/scala/com/lookout/borderpatrol/RoutingFilter.scala [hunk 1]
```diff
@@ -1,29 +1,30 @@
...
+    val rRequest = RoutedRequest(request, serviceName(request.getUri))
+    println( "Session type " + rRequest.session)
+    println(" Service is >" + rRequest.service + "<")
+    val r = service(rRequest)
     println("<----------------------------- RoutingFilter ------------------------------")
     r
   }
+
+  //TODO: use pattern matching + Regex
+  def serviceName(uri: String): String = uri match {
```
Comment (wkimeria) [inline borderpatrol-core/src/main/scala/com/lookout/borderpatrol/RoutingFilter.scala]: This needs to be fixed. It should only look at the first part of the uri and should be looking for hard coded

File: borderpatrol-core/src/main/scala/com/lookout/borderpatrol/BorderPatrolApp.scala [hunk 2]
```diff
@@ -1,55 +1,102 @@
...
+
+    val router2 = RoutingService.byPath[HttpRequest] {
+      case "/foo" => new FooService(" group 2")
+      case "/foo/" => new FooService(" group 2")
+      case "/foo/good" => new FooService(" group 2")
+      case "/a" => authPipeline
+      case "/a/" => authPipeline
+    }
+
+    val server2 = ServerBuilder()
```
Comment (wkimeria) [inline borderpatrol-core/src/main/scala/com/lookout/borderpatrol/BorderPatrolApp.scala]: This needs to be removed. This was me playing around with testing load balancing, and is not used

File: borderpatrol-core/src/main/scala/com/lookout/borderpatrol/UpstreamService.scala [hunk 3]
```diff
@@ -1,36 +1,55 @@
...
+  def apply(request: RoutedRequest) = {
+    println("------------------------------ UpstreamService " + request.getUri + "----------------------------->")
+
+    //TODO: Rewrite url
+    val originalUri = request.getUri
+    //val mappedUrl = getRewrittenUrl(originalUri)
+    val mappedUrl = originalUri
+
 
+    //TODO: Change this to take a routed request in order to get the service name. HardCoded for now
```
Comment (wkimeria) [inline borderpatrol-core/src/main/scala/com/lookout/borderpatrol/UpstreamService.scala]: This todo no longer applies. Remove

**Comment by trane** [comment]
killing this, @wkimeria will be issuing a new PR with the changes he noted.

**Comment by trane** [comment]
re-opening for will to merge

**Pull request merged**